### ISSUE TITLE
Distributed training (multi-GPU) error on TITAN RTX

### ISSUE THREAD

**Issue by Leslie0115** [tianzhi0549/FCOS#247]
Hi, I use two TITAN RTX for training, the program is stuck in "INFO: Start training" but no longer responding. By "nvidia-smi" checking GPU status, memory usage of GPU-0 is about 2G and volatile GPU-Util is 100%, while memory usage of GPU-1 is about 700M and volatile GPU-Util is 0%. When I use single GPU for training that everything is normal and do not encounter this problem. I have used "torch.nn.DataParallel" for multi-GPU training on other projects and it runs normally, so I guess if it is because TITAN RTX is not compatible with "nn.parallel.DistributedDataParallel", have you ever encountered the same problem or can you provide some suggestions? Thanks very much.

**Comment by tianzhi0549** [comment]
@Leslie0115 What changes did you make to the project? It seems a problem related to `NCCL`.

**Comment by Leslie0115** [comment]
@tianzhi0549 I made a lot of changes, but I should not modify anything related to distributed read or write. I found someone saying that RTX's read or write protection does have some problems with "nn.parallel.DistributedDataParallel", so I wait and see if the update of pytorch can solve this problem, or I use "torch.nn.DataParallel" for multi-GPU training.

**Comment by Leslie0115** [comment]
@tianzhi0549 By the way  I want to ask another question, I tried to modify this project to achieve rotating target detection, I modify the box_pred regressions to (l, t, r, b, theta), theta refers to the angle of bbox, and I use smooth L1 loss in log scale for training. However, after about 60 iterations, the reg_loss becomes nan, I haven't found the reason so far. Could you give me some tips or suggestions? Thanks a lot.

**Comment by tianzhi0549** [comment]
@Leslie0115 1) I would suggest that you train a model with exactly the original code to see what will happen. 2) Please try to clip the gradients during training.

**Comment by Leslie0115** [comment]
@tianzhi0549 I found that modifying the **_initial method_** can solve the problem of distributed training on TITAN RTX. The "init_method" param of "torch.distributed.init_process_group()" in the source code is "env://" while encountering the problem, and when changing this param to "tcp://" it will be OK (of course need to make some changes accordingly about the initial method).
I will close this issue.
